visual-dotnet/SauceLabs.Visual/VisualClient.cs

```diff
@@ -91,15 +99,24 @@ private async Task SetupBuild(CreateBuildOptions buildOptions)
                     throw new VisualClientException($"build {build.Id} is not RUNNING");
                 }
                 Build = build;
-                _externalBuild = true;
+                Build.IsExternal = true;
+                _sharedBuild = Build;
+                return;
             }
-            else
+
+            buildOptions.CustomId ??= EnvVars.CustomId;
+            var createBuildResponse = await CreateBuild(buildOptions);
+            Build = new VisualBuild(createBuildResponse.Id, createBuildResponse.Url, createBuildResponse.Mode)
             {
-                buildOptions.CustomId ??= EnvVars.CustomId;
-                var createBuildResponse = await CreateBuild(buildOptions);
-                Build = new VisualBuild(createBuildResponse.Id, createBuildResponse.Url, createBuildResponse.Mode);
-                _externalBuild = false;
-            }
+                IsExternal = false
+            };
+            var copiedApi = _api.Clone();
```

Comment:
based on the design we're allowing to send data to a different instances (based on region credentials etc). We're giving this impression to our customers based on our API design. In the end it doesn't matter because the API is also cloned, so their changes are not respected.

How about leaving the API client dedicated for each visual client?

Reply:
Leaving the API client dedicated for each visual will generate probably more issues.
We expect users to dispose of `VisualClient` after the class has been processed. That will dispose both the `VisualApi` instance and it's internal `GraphQLClient`. Cloning the `VisualApi` itself, ensure that it it still valid when it needs to be invoked.